src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -3159,3 +3178,76 @@ void Pane::CollectTaskbarStates(std::vector<winrt::TerminalApp::TaskbarState>& s
         _secondChild->CollectTaskbarStates(states);
     }
 }
+
+void Pane::EnableBroadcast(bool enabled)
+{
+    if (_IsLeaf())
+    {
+        _broadcastEnabled = enabled;
+        UpdateVisuals();
+    }
+    else
+    {
+        _firstChild->EnableBroadcast(enabled);
+        _secondChild->EnableBroadcast(enabled);
+    }
+}
+
+void Pane::BroadcastKey(const winrt::Microsoft::Terminal::Control::TermControl& sourceControl,
+                        const WORD vkey,
+                        const WORD scanCode,
+                        const winrt::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                        const bool keyDown)
+{
+    if (_IsLeaf())
+    {
+        if (_control != sourceControl && !_control.ReadOnly())
+        {
+            _control.BroadcastKeyEvent(vkey, scanCode, modifiers, keyDown);
+        }
+    }
+    else
+    {
+        _firstChild->BroadcastKey(sourceControl, vkey, scanCode, modifiers, keyDown);
+        _secondChild->BroadcastKey(sourceControl, vkey, scanCode, modifiers, keyDown);
+    }
+}
+
+void Pane::BroadcastChar(const winrt::Microsoft::Terminal::Control::TermControl& sourceControl,
+                         const wchar_t character,
+                         const WORD scanCode,
+                         const winrt::Microsoft::Terminal::Core::ControlKeyStates modifiers)
+{
+    if (_IsLeaf())
+    {
+        if (_control != sourceControl && !_control.ReadOnly())
+        {
+            _control.BroadcastChar(character, scanCode, modifiers);
+        }
+    }
+    else
+    {
+        _firstChild->BroadcastChar(sourceControl, character, scanCode, modifiers);
+        _secondChild->BroadcastChar(sourceControl, character, scanCode, modifiers);
+    }

```

Comment:
nit: we've been moving away from manual tree walks and towards `WalkTree<T>`. We should do that for all of these as well.